### PR TITLE
Hotfix: Drop final field from build id

### PR DESCRIPTION
**Pull request by krivard**
# Overview

## What problem does this address?

We dropped the final field from the build id in hotfix #5404, but the build resolver still expected that field in its patterns.

## What did you change?

Drop final field from build resolver patterns.

# Testing

How did you make sure this worked? How can a reviewer verify this?

In a notebook, `get_build_from_tag("v2026.7.1")` succeeds.

## To-do list
- [x] Review the PR yourself and call out any questions or issues you have.

### CODE REVIEW

File: src/pudl/deploy/pudl.py [hunk 1]
```diff
@@ -621,7 +623,7 @@ def get_build_from_tag(tag: str) -> UPath:
     # Check that we found a build
     if most_recent_build_path is None:
         raise RuntimeError(
-            f"Can't find a build associated with tag: {tag}, ref: {git_ref}"
+            f"Can't find a build associated with tag: {tag}, ref: {git_ref}; checked:\n{'\n'.join(checked)}"
```
Comment (krivard) [inline src/pudl/deploy/pudl.py:626]: this extra "checked" is my standard "if I saw this error what is the next information i would immediately want, and is that something the code already has easier access to than future-debugging-me will have?" change

File: tests/unit/deploy/deploy_pudl_test.py
Comment (krivard) [inline tests/unit/deploy/deploy_pudl_test.py:1]: there are other build ids with the extra field in this test file but none of them go through `get_build_from_tag`. pushing those to the cleanup pass post-release

**Review by zaneselvans — APPROVED** [review]
